src/compiler/checker.ts

```diff
@@ -3178,7 +3178,7 @@ namespace ts {
                 result.pattern = pattern;
             }
             if (hasComputedProperties) {
-                result.flags |= TypeFlags.ObjectLiteralPatternWithComputedProperties;
+                result.inObjectLiteralPatternWithComputedProperties = hasComputedProperties;

```

Comment:
Also, shouldn't it be `isObjectBlaBlaBla...`? In other words, "is" instead of "in".